### PR TITLE
tools: create script to generate lp release branches (SC-448)

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Notes
This turns what is currently an inline script in the RELEASES.md instructions into a separate script that can be run as part of our release process.

## Proposed Commit Message

```
tools: create script to generate lp release branches
```

## Test Steps

Try running the new script.

## Desired commit type::

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [ ] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist:

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: tools/create-lp-release-branches.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  set -x
+  set -e
+fi
+
+for release in xenial bionic focal hirsute
+do
+  echo
+  echo $release
+  if [ -z "$DO_IT" ]; then
+    echo git checkout upload-${UA_VERSION}-${DEVEL_RELEASE} -B upload-${UA_VERSION}-$release
```
Comment (lucasmoura) [inline tools/create-lp-release-branches.sh:40]: I think we can store the command in a variable and just reference it on both situations:
```bash
cmd="git checkout upload-${UA_VERSION}-${DEVEL_RELEASE} -B upload-${UA_VERSION}-$release"
if [ -z "$DO_IT" ]; then
  echo $cmd
else:
  $cmd
fi
```

Comment (orndorffgrant) [reply tools/create-lp-release-branches.sh:40]: Yes good point!

**Comment by orndorffgrant** [comment]
@lucasmoura updated!

File: tools/create-lp-release-branches.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    $checkout_cmd
+  fi
+
+  case "${release}" in
+      xenial) version=${UA_VERSION}~16.04.1;;
+      bionic) version=${UA_VERSION}~18.04.1;;
+      focal) version=${UA_VERSION}~20.04.1;;
+      hirsute) version=${UA_VERSION}~21.04.1;;
+  esac
+  dch_cmd=(dch -v ${version} -D ${release} -b  "Backport new upstream release: (LP: #${SRU_BUG}) to $release")
```
Comment (lucasmoura) [inline tools/create-lp-release-branches.sh:53]: We are treating this as an array because of the changelog message in the middle of the command ?

**Comment by renanrodrigo** [comment]
Just one question: do we want to keep dry-run as the standard? No strong opinion on this, just asking because usually it's the opposite.

File: tools/create-lp-release-branches.sh (hunk 2 above)
Comment (orndorffgrant) [reply tools/create-lp-release-branches.sh:53]: Correct! The quotes get inserted as part of the changelog otherwise.

**Comment by orndorffgrant** [comment]
@renanrodrigo Good point! I did the opposite to naturally guide someone through using it. So if the release process can just say "go run this script".
Then [insert new hire who is unfamiliar or me 6 months from now who has forgotten how this works] can just run the script. They'll see the messages to use the envvars. Then after adding the envvars, they'll see the dry run output and can see what its doing. Then the script tells them to add the do-it var, and then it all works.

I'm totally open to changing it though if we prefer a different approach

**Review by lucasmoura — APPROVED** [review]

**Review by renanrodrigo — APPROVED** [review]

**Comment by renanrodrigo** [comment]
@orndorffgrant makes sense - good point of view

**Comment by orndorffgrant** [comment]
Just a flaky lxd issue on jenkins. Merging

**Pull request merged**